src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -179,6 +183,36 @@ def sort_code_mapping_configs(
                     sorted_configs.insert(0, config)
         return sorted_configs
 
+    def get_codecov_path(self, source_url: str) -> Optional[str]:
+        # Extract path from URL from "blob/<sha_or_branch>/" onwards
+        blob_index = source_url.find("blob")
+        if blob_index != -1:
+            codecov_path_index = source_url.find("/", blob_index + len("blob/")) + len("/")
+            return source_url[codecov_path_index:]
+        return None
+
+    def get_codecov_line_coverage(
+        self, repo: str, service: str, branch: str, source_url: str, access_token: str
+    ) -> Tuple[Optional[Sequence[Tuple[int, int]]], Optional[int]]:
+        owner_username, repo_name = repo.split("/")
+        url = CODECOV_URL.format(
+            service=service, owner_username=owner_username, repo_name=repo_name
+        )
+        path = self.get_codecov_path(source_url)
+        line_coverage, status_code = None, None
+        if path:
+            params = {"branch": branch, "path": path}
+            response = requests.get(
+                url, params=params, headers={"Authorization": "tokenAuth %s" % access_token}
+            )
+            line_coverage, status_code = (
+                response.json()["files"][0]["line_coverage"],
+                response.status_code,
+            )
+        else:
+            logger.error("Could not find codecov path")
```

Comment:
`path` will be part of the signature with my recommended changes and you will be able to remove this.